pkg/infra/httpclient/httpclientprovider/http_client_provider.go

```diff
@@ -50,15 +57,76 @@ func New(cfg *setting.Cfg, validator models.PluginRequestValidator, tracer traci
 				return
 			}
 			datasourceLabelName, err := metricutil.SanitizeLabelName(datasourceName)
-
 			if err != nil {
 				return
 			}
+
+			_, enableSecureSocksProxy := opts.CustomOptions["secure_socks_proxy"]
+			if cfg.IsFeatureToggleEnabled(featuremgmt.FlagSecureSocksDatasourceProxy) && enableSecureSocksProxy {
+				err = newSecureSocksProxy(cfg, transport)
+				if err != nil {
+					logger.Error("Failed to enable the socks proxy", "error", err.Error(), "datasource", datasourceName)
+				}
+			}
+
 			newConntrackRoundTripper(datasourceLabelName, transport)
 		},
 	})
 }
 
+// newSecureSocksProxy takes a http.DefaultTransport and wraps it in a socks5 proxy with TLS
+func newSecureSocksProxy(cfg *setting.Cfg, transport *http.Transport) error {
```

Comment:
Nit. Would consider moving this to a separate file. In addition, seems like the function could take `setting.SecureSocksDSProxySettings` instead of `*settings.Cfg`

Reply:
Good suggestion; I moved it and updated the parameter.